_posts/2020-05-18-mirror-kafka-eventhub.md

```diff
@@ -0,0 +1,225 @@
+
+Assuming that the Apache Kafka topic to mirror is named `testeh`, we have to create a corresponding Event Hub in a related namespace.
+It is possible to do that following the official Microsoft documentation, using the [Azure portal](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-create), the [Azure CLI](https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-quickstart-cli), Azure Powershell or ARM template.
+
+![Azure Portal: Event Hub creation](/assets/images/posts/2020-05-18-azure-portal-eventhub.png)
+
+Having the Event Hub ready is just the first step.
+In order to configure Kafka Mirror Maker for connecting to the Event Hub, we need to get the corresponding connection string for being authenticated and authorized to write on the Event Hub itself.
+Just for simplicity, we can use the already provided `RootManageSharedAccessKey` policy that enable all the rights to manage, send and listen in the Event Hub namespace in which in the `testeh` lives.
+In a real scenario, we would like to create a policy for the mirroring part in order to enable the Kafka Mirror Maker producer just to write to the Event Hub while using a different policy for the application consuming the messages with just the listen right.
+
+The only needed information from the above policy is the connection string which has the following format:
+
+```
+Endpoint=sb://<eventhubs-namespace>.servicebus.windows.net/;SharedAccessKeyName=RootManageSharedAccessKey;SharedAccessKey=<access-key>
+```
+
+Where `<eventhubs-namespace>` has to be replaced with the actual Event Hub namespace you are going to use.
+
+![Azure Portal: Event Hub connection string](/assets/images/posts/2020-05-18-connectionstring-eventhub.png)
+
+### Configure Kafka Mirror Maker
+
+When mirroring messages from a source Apache Kafka cluster to a target one, the usual paradigm is deploying in the Kafka Mirror Maker instance alongside the target cluster.
+It means consuming remotely and producing locally.
+In this integration scenario, it is not possible because there is no way to deploy Kafka Mirror Maker alongside the Event Hub namespace that is the target cluster.
+For this reason, the paradigm will be slightly different, deploying Kafka Mirror Maker alongside the source Apache Kafka cluster where we have more control.
+Anyway, it's worth mentioning that the Kubernetes cluster, where Apache Kafka runs, could be an AKS (Azure Kubernetes Service) one so as part of the same Microsoft Cloud infrastracture.
+
+The first step is storing the Event Hub authentication info into a Kubernetes `Secret` in order to be referenced in the Strimzi `KafkaMirrorMaker` custom resource for enabling the producer part to connect to the Event Hub itself.
+The following snippet shows such a `Secret` where the `$ConnectionString` hasn't to be changed while the endpoint has to be customized with the actual Event Hub connection string; save this `Secret` in a file named `eventhubs-secret.yaml`
```

Comment:
`$ConnectionString`? I see no `$ConnectionString`. Also "hasn't to be changed" → "must not be changed" or "need not be changed"

Reply:
I removed a configuration from the snippet but forgot to remove the mention here. Good catch!